core-bundle/src/Security/TwoFactor/BackupCodeManager.php

```diff
@@ -42,12 +48,18 @@ public function invalidateBackupCode($user, string $code): void
             return;
         }
 
-        $backupCodes = json_decode($user->backupCodes, true);
-        $key = array_search($code, $backupCodes, true);
+        $key = false;
+        $backupCodes = array_values(json_decode($user->backupCodes, true));
+
+        foreach ($backupCodes as $key => $backupCode) {
+            if (password_verify($code, $backupCode)) {
+                break;
+            }
+        }
```

Comment:
If no one of the backup codes matches, `$key` would be set to the last index and would be invalidated. This is probably not what we want?

Reply:
True story! 🙃

Reply:
I've changed it in b1f496377.